### DISCUSSION TITLE
Injecting securityContext into dynamically created Sensor/EventSource deployments

### DISCUSSION THREAD

**Discussion by bishalkr** [argoproj/argo-events#3330] [Q&A]
I am currently facing an issue with Sensor and EventSource deployments in Argo Events. In my K8s cluster, we have OPA in place, which requires securityContext to be set for all pods. However, the pods created by the Event Controller for Sensor and EventSource do not include the necessary securityContext, which prevents them from being scheduled.

Is there a way to configure Argo Events so that a securityContext is injected into the Sensor and EventSource deployments automatically? Ideally for my use case, whenever a new Sensor or EventSource is created, its corresponding deployment should have the securityContext injected.

From what I understand, these deployments are handled by the Event Controller, but I haven't found any configuration options to modify the security settings for Sensor or EventSource deployments.

Any help or guidance on this would be greatly appreciated!

Thanks

**Comment by smashingraph** [comment]
Hi @bishalkr,

As you are using OPA, you may have a look at https://open-policy-agent.github.io/gatekeeper/website/docs/mutation/
You could use it to automatically inject what you need,

Best Regards,
